core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -185,24 +185,24 @@ trait AttributeComponent {
         we could modify this function. It would need to inspect, or look up, some flag on the workspace
         that indicates whether or not the workspace is sharded.
 
-        Sharded workspaces would use a shard identifier such as "a_07", while unsharded workspaces
-        would use a shard identifier of "unsharded". This would translate into referencing tables named
-        ENTITY_ATTRIBUTE_a_07 or ENTITY_ATTRIBUTE_unsharded, respectively.
-
-        In such a case, we'd also need to update the liquibase changeset, which currently renames ENTITY_ATTRIBUTE to
-        ENTITY_ATTRIBUTE_archive; we'd want to rename it to ENTITY_ATTRIBUTE_unsharded. We'd also need to update
-        liquibase to not copy all attribute rows into the shards; we'd only want to copy rows for any workspaces
-        that should use sharding.
+        Sharded workspaces would use a shard identifier such as "04_07", while unsharded workspaces
+        would use a shard identifier of "archive". This would translate into referencing tables named
+        ENTITY_ATTRIBUTE_04_07 or ENTITY_ATTRIBUTE_archive, respectively.
      */
 
-    // see the liquibase changeset "20210615_sharded_entity_tables.xml" for expected shard identifier values
+    val shardSize = 4 // range of characters in a shard, e.g. 4 = "00-03"
```

Comment:
Does this need to change to 5 or am I misunderstanding? (I've never actually sharded a db so I'm learning as I go!)

Reply:
this value is really only used in our custom code, it's used to figure out how to go from workspaceId -> shard name. Our algorithm looks at the first two characters of the workspaceId. A value of "4" here means that we include 4 consecutive values in each shard, like this:

shard 1:
* 00xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 01xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 02xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 03xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

shard 2:
* 04xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 05xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 06xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
* 07xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

Reply:
ahhh, got it!